contribute-to-tidb/review-a-pr.md

```diff
@@ -7,46 +7,163 @@ This guide is for all contributors that want to help with reviewing code contrib
+
+### Complexity
+
+Is the PR more complex than it should be? Check this at every level of the PR—are individual lines too complex? Are functions too complex? Are classes too complex? “Too complex” usually means “can’t be understood quickly by readers.” It can also mean “developers are likely to introduce bugs when they try to call or modify this code.”
+
+A particular type of complexity is over-engineering, where developers have made the code more generic than it needs to be, or added functionality that isn’t presently needed by the system. Reviewers should be especially vigilant about over-engineering. Encourage developers to solve the problem they know needs to be solved now, not the problem that the developer speculates might need to be solved in the future. 
+
+### Tests
+
+Ask for unit, integration, or end-to-end tests as appropriate for the change. In general, tests should be added in the same PR as the production code.
+Make sure that the tests in the PR are correct, sensible, and useful. Are they covering interesting cases? Are they readable? Does the PR lower overall test coverage? 
+Think of ways this code could break. Will the tests actually fail when the code is broken? If the code changes beneath them, will they start producing false positives? Does each test make simple and useful assertions? Are the tests separated appropriately between different test methods?
+
+Remember that tests are also code that has to be maintained. Don’t accept complexity in tests just because they aren’t part of the main binary.
+Last but not least, keep an eye on how long the single unit takes to run. Does it slow the CI?
+
+### Naming
+
+Did the developer pick good names for everything? A good name is long enough to fully communicate what the item is or does, without being so long that it becomes hard to read.
+
+### Comments
+
+Did the developer write clear comments in understandable English? Are all of the comments actually necessary? Usually comments are useful when they explain why some code exists, and should not be explaining what some code is doing. If the code isn’t clear enough to explain itself, then the code should be made simpler. There are some exceptions (regular expressions and complex algorithms often benefit greatly from comments that explain what they’re doing, for example) but mostly comments are for information that the code itself can’t possibly contain, like the reasoning behind a decision.
+It can also be helpful to look at comments that were there before this PR. Maybe there is a TODO that can be removed now, a comment advising against this change being made, etc.
+Note that comments are different from documentation of classes, modules, or functions, which should instead express the purpose of a piece of code, how it should be used, and how it behaves when used.
```

Comment:
```suggestion
* Does the developer write clear comments in understandable English?
* Are all of the comments actually necessary?

Usually comments are useful when they explain why the code exists, while not explain what the code is doing. If the code isn't clear enough to explain itself, then the code should be made simpler.

There are some exceptions; for example, regular expressions and complex algorithms often benefit greatly from comments that explain what they’re doing. But mostly comments are for information that the code itself can hardly contain, like the reason behind a decision.

It can also be helpful to look at comments existing before this PR. Maybe there is a `TODO` can be removed now, a comment advising against this change being made, etc.

Note that comments are different from documentation of classes, modules, or functions, which should instead express the purpose of a piece of code, how it should be used, and how it behaves when used.
```